preview-thumbnails via src:callback()

doublex opened the pull request:
I have a request: We can't use the current VTT-`previewThumbnails` for some reasons.
Would you mind to extend the API with a callback (see the attached pull request)?

E.g.:
```` 
previewThumbnails: {
    enabled: true,
    src: function() {
        return [{
            frames: [{
                startTime: 0,
                endTime: 1000,
                text: "http://localhost/demo/240p-00001.jpg",
                x: 0,
                y: 0,
                w: 427,
                h: 240},
                //...
            ],
            width: 2135,
            height: 1200,
            urlPrefix: "",
        }];
    },
}
````

sampotts commented:
Thanks for looking at this 👍 This PR will fail ESLint though. Please check you have ESLint installed and running. Preferably with Prettier. 

- We don't use underscores in variable names. Please stick to camelCase. i.e. you could just call the function `resolve`. 
- We have built-in type checking helpers. i.e. `is.function(...)`
- If statements need a space after the if and no spaces inside the parenthesis - i.e. `if (something) {`

doublex commented:
Fixed. And I have changed the API e.g. to call an XHR request in the callback:
Thanks a lot for your help.

````javascript
previewThumbnails: {
    enabled: true,
    src: function(resolver) {
        var thumbnails = [{
            frames: [{
                startTime: 0,
                endTime: 10000,
                text: "http://localhost/demo/240p-00001.jpg",
                x: 0,
                y: 0,
                w: 427,
                h: 240},
                //...
            ],
            width: 2135,
            height: 1200,
            urlPrefix: "",
        }];
        resolver(thumbnails);
    },
}
````

doublex commented:
Thanks a lot for your help!
